feat: gemma-3-1b-it 모델을 이용한 시 출력하기

wibaek opened the pull request:
1b 모델로 전환했습니다!

wibaek committed:
gemma-3-1b-it 모델을 이용한 시 출력하기

yoonc01 commented on main.py:
```diff
@@ -0,0 +1,34 @@
+    ],
+]
+
+output = pipe(messages, max_new_tokens=50)
```
핑프를 위해 아래와 같이 주석을 다는 거는 어떤가용??

```js
// output 형태
[
    [
        {
            'generated_text': [
                {
                    'role': 'system',
                    'content': [{'type': 'text', 'text': 'You are a helpful assistant.'}]
                }, 
                {
                    'role': 'user',
                    'content': [{'type': 'text', 'text': '소프트웨어 마에스트로에 관련된 시를 써줘'}]
                },
                {
                    'role': 'assistant',
                    'content': '## 소프트웨어 마에스트로\n\n어둠 속, 코드의 춤\n새로운 세계를 창조하는 예술가\n소프트웨어 마에스트로, 그 이름은\n세상의 지식으로 빚어낸 섬세'
                }
            ]
        }
    ]
]
```

yoonc01 commented on main.py:
```diff
@@ -0,0 +1,34 @@
+    torch_dtype=torch.bfloat16,
+)
+
+messages = [
+    [
+        {
+            "role": "system",
+            "content": [
+                {"type": "text", "text": "You are a helpful assistant."},
+            ],
+        },
+        {
+            "role": "user",
+            "content": [
+                {"type": "text", "text": "소프트웨어 마에스트로에 관련된 시를 써줘"},
+            ],
+        },
+    ],
+]
```
사실 [공식문서 사용법](https://huggingface.co/google/gemma-3-1b-it#running-with-the-pipeline-api)에서는 이러한 형식을 쓰라고 하긴 했는데 

```python
messages = [
    {
        "role": "system",
        "content": [{"type": "text", "text": "You are a helpful assistant."}]
    },
    {
        "role": "user",
        "content": [{"type": "text", "text": "소프트웨어 마에스트로에 관련된 시를 써줘"}]
    }
]
```

불경한 마음으로 이렇게도 넣어봤는데 돌아가걸랑??? 굳이 저 포멧을 공식문서에서 쓴 이유가 있을랑가??

sihyeonn replied:
formatting 차이여서 어떤 방식을 써도 상관 없어요!

wibaek replied:
black 포매터를 써서 자동 포매팅 된 것 같네요😁

yoonc01 commented:
수고하셨습니다~

sihyeonn commented on pyproject.toml:
```diff
@@ -0,0 +1,10 @@
+[project]
+name = "pythonicml"
+version = "0.1.0"
+description = "Add your description here"
```
넣는 김에 README.md도 업데이트하고, description도 적어주세요~

sihyeonn commented on pyproject.toml:
```diff
@@ -0,0 +1,10 @@
+version = "0.1.0"
+description = "Add your description here"
+readme = "README.md"
+requires-python = ">=3.13"
```
requires-python이 너무 높은 것 같아요!
효준님 문제 없으면 일단 두고, 나중에 python version 여러개 넣고 호환성 테스트 하는 것도 해보면 좋겠네요

sihyeonn commented on main.py:
```diff
@@ -0,0 +1,34 @@
+
+# https://huggingface.co/google/gemma-3-1b-it
+ACCESS_TOKEN = "HUGGINGFACE_ACCESS_TOKEN_COMES_HERE"
+login(ACCESS_TOKEN)
```
login하면 실패처리 이런 것도 필요할 것 같아요
os.environ으로 설정하는 것도 괜찮을 것 같습니다!

sihyeonn commented on main.py:
```diff
@@ -0,0 +1,34 @@
+
+pipe = pipeline(
+    "text-generation",
+    model="google/gemma-3-1b-it",
```
모델도 위에 상수로 명시하고 바꿔가면서 테스트 할 수 있음 좋을 것 같아요!

sihyeonn reviewed (approved):
생각해봤는데, python이랑 rust latency 성능을 비교하려면 model이 시간이 오래걸리면 안될 것 같아요
모델이 python 성능보다 latency가 낮은 경우에는 모델 성능에 좌우될 것이라 backend server 자체에 영향이 적을 것 같아서요!
그래서 속도가 빠른 경량화된 모델로 해보는 것이 좋을 듯 합니다
예로, 긍정부정 판단 같은 작은 모델이용

wibaek committed:
chore: Python version 3.13 -> 3.12

호환성 검증 완료

uv.lock을 .gitignore에 넣지 않아도 될까? -> https://velog.io/@ldhbenecia/Dependency-Management-in-uv-A-Developers-Guide

wibaek committed:
refactor: 모델 상수로 분리, 예상 아웃풋 주석

wibaek committed:
chore: add description in pyproject.toml and update README.md

wibaek committed:
refactor: 허깅페이스 토큰을 환경변수로 받도록 수정, 로그인 오류시 종료 처리

sihyeonn commented on main.py:
```diff
@@ -0,0 +1,68 @@
+ACCESS_TOKEN = os.environ["HF_TOKEN"]
+
+try:
+    login(ACCESS_TOKEN)
```
token있음 login 안해도 될거여요 아마도..

wibaek replied:
그렇네요😊

wibaek committed:
feat: 로그인 코드 삭제
